Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -35,7 +41,23 @@ class ChannelDTO: NSManagedObject {
     @NSManaged var messages: Set<MessageDTO>
     @NSManaged var reads: Set<ChannelReadDTO>
     @NSManaged var attachments: Set<AttachmentDTO>
-    
+
+    override func willSave() {
+        super.willSave()
+
+        // Change to the `trunctedAt` value have effect on messages, we need to mark them dirty manually
+        // to triggers related FRC updates
+        if changedValues().keys.contains("truncatedAt") {
+            messages
+                .filter { !$0.hasChanges }
+                .forEach {
+                    // Simulate an update
+                    $0.willChangeValue(for: \.id)
+                    $0.didChangeValue(for: \.id)
+                }
+        }
+    }
```

Comment:
I didn't understand what's going on here... What are the updates we need to trigger?

Reply:
I created CIS-695 which improves this. It's the same trick we use for attachments and messages. Fetched results controller doesn't observe changes in related objects, even when you specify them in the predicate. So for example:
```
predicate = ALL messages WHERE message.createdAt > channel.truncatedAt
```
will get evaluated properly when you fetch the first time. However, when you change the `channel.truncatedAt` value, the FRC ignores it because it observers only changes to `MessageDTO` entity. This way we force the FRC to re-evaluate the predicate and update the results.